scala/private/phases/phase_merge_jars.bzl

```diff
@@ -0,0 +1,19 @@
+#
+# PHASE: merge jars
+#
+# DOCUMENT THIS
+#
+load(
+    "@io_bazel_rules_scala//scala/private:rule_impls.bzl",
+    "merge_jars",
+)
+
+def phase_merge_jars(ctx, p):
```

Comment:
In general, I want to avoid too much of renaming and refactoring to make this PR easier for review. I agree we should name it after it's responsibility and also we should move the helper functions into these phases (no need for a call to other files).

I like the idea but how about we handle it in another PR as well?